### ISSUE TITLE
limit misleading? 

### ISSUE THREAD

**Issue by wintech147** [MicrosoftDocs/office-docs-powershell#6099]
The parameter for -FileData mentions there is a max file size of 100kb however this should show or warn that this is a tenant wide limit. If you try to upload multiple keyword dictionary lists that all together go over the 100kb size limit you can no longer upload dictionary lists to your tenant. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 12df6f61-8d11-f1cc-897a-14ee372e8c29
* Version Independent ID: 46f7615a-a8e0-4e24-64eb-26815c40dcfa
* Content: [Set-DlpKeywordDictionary (exchange)](https://docs.microsoft.com/en-us/powershell/module/exchange/set-dlpkeyworddictionary?view=exchange-ps)
* Content Source: [exchange/exchange-ps/exchange/Set-DlpKeywordDictionary.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/exchange/exchange-ps/exchange/Set-DlpKeywordDictionary.md)
* Service: **exchange-powershell**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by yogkumgit** [comment]
@officedocsbot assign @yogkumgit

**Comment by dariomws** [comment]
Hi @wintech147, thank you for your feedback and help us to improve docs.microsoft.com.

Could you kindly help me to understand how you get to this conclusion? I'm testing it and get the following error only when the keywords list is big enough:
"Sending data to a remote command failed with the following error message: The current deserialized object size of the data
received from the remote client computer exceeded the allowed maximum object size. The current deserialized object size is
10572800. The allowed maximum object size is 10485760. For more information, see the about_Remote_Troubleshooting Help topic."

Thank you so much!

**Comment by wintech147** [comment]
Hi @dariomws thanks for reaching out. Hopefully I can explain clearly :) My customer has multiple custom keyword dictionary lists they want to leverage for DLP. Some of them are fairly large (over 100kb limit) which they saw via the GUI and via PowerShell when trying to do their initial upload. They then split them up so they each were under the 100kb limit per file. 

My customer then started uploading the keyword lists and were able to get 5 of them uploaded before receiving the same "100kb" limit error even though the file itself was only 75kb. After working with O365 support they informed my customer that it's 100kb at the tenant not just per file. So after uploading the multiple lists, they all together added up to 100kb thus keeping them from uploading any additional custom keyword lists. 

This feedback could also be leveraged for the New-DlpKeywordDictionary command as well. And misleading might have been the wrong word to use, my apologies, I simply mean that noting the 100kb limit is a tenant limit and not necessarily per file would be hugely helpful. I hope this made sense :)

**Comment by dariomws** [comment]
@wintech147 thank you so much for your explanation. 

Unfortunately, I cannot reproduce this behavior. I can upload two text files of almost 100 KB. I think it takes into account the size of the encoded keywords using:
$EncodedKeywords = $Keywords | ForEach-Object{[System.Text.Encoding]::UTF8.GetBytes($_+"`r`n")};

Even in this way, I could upload a file of 121 KB. I'll try to get more information.

**Comment by wintech147** [comment]
Hi @dariomws ah my mistake I forgot to mention one thing! :) We were able to ultimately upload roughly 221kb of data as well before hitting a point where we were no longer able to upload. According to the support engineer there is compression done on the tenant side so that even though we saw 221kb of data the actual size was within a few kb's of the tenant limit (100kb give or take). 

In my demo tenant I took the customers same lists and saw the same experience where it let me go for a few lists (as long as each lists was under 100kb) but then was no longer able to upload keyword lists after about 5 of the previous lists uploading etc.
